### PR TITLE
Miscellaneous fixes for building 1.1.1 tests against 3.0 libraries

### PR DESCRIPTION
**Pull request by mattcaswell**
I have detailed notes on all of these changes some of which are going to turn into "issues". Some issues already raised are https://github.com/openssl/openssl/issues/14370, https://github.com/openssl/openssl/issues/14379, https://github.com/openssl/openssl/issues/14411

### CODE REVIEW

**Comment by mattcaswell** [comment]
I had a few more but we seem to have done some double work!

File: test/recipes/30-test_evp_data/evpmac.txt [hunk 1]
```diff
@@ -131,23 +131,29 @@ Output = 5150d1772f50834a503e069a973fbd7c
 # SIPHASH - default values: 2,4 rounds, explicit 8-byte mac
 
 MAC = SipHash
-Ctrl = digestsize:8
+#In 3.0 this ctrl seems to have been renamed to "size"
```
Comment (levitte) [inline test/recipes/30-test_evp_data/evpmac.txt:134]: Perhaps we should ensure that there's an alias in 3.0...  Not gonna block this change, though...

File: test/evp_test.c [hunk 2]
```diff
@@ -1997,7 +1997,8 @@ static int keygen_test_run(EVP_TEST *t)
     t->err = NULL;
     if (EVP_PKEY_keygen(keygen->genctx, &pkey) <= 0) {
         t->err = "KEYGEN_GENERATE_ERROR";
-        goto err;
+        EVP_PKEY_free(pkey);
```
Comment (levitte) [inline test/evp_test.c]: Wait what???  This seems wrong, it's absurd to expect that a failing EVP_PKEY_keygen() produces an EVP_PKEY.  I'd say this is a bug in 3.0!

**Review by levitte — CHANGES_REQUESTED** [review]

File: test/evp_test.c (hunk 2 above)
Comment (mattcaswell) [reply test/evp_test.c]: Oh...this might not actually be necessary

Comment (mattcaswell) [reply test/evp_test.c]: The important point was the change in return value (0 -> 1)

Comment (mattcaswell) [reply test/evp_test.c]: 0 means "I wasn't even able to run the test", 1 means "I ran the test successfully (but the thing I tried to do might have failed - and that could be ok and expected)"

File: test/recipes/30-test_evp_data/evpmac.txt (hunk 1 above)
Comment (mattcaswell) [reply test/recipes/30-test_evp_data/evpmac.txt:134]: This is a bug in 3.0 IMO. It's being tracked through https://github.com/openssl/openssl/issues/14370, this is a temporary workaround to make the tests pass until we fix the bug.

**Comment by mattcaswell** [comment]
Fixup pushed addressing the feedback.

**Pull request merged**